node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -497,19 +502,32 @@ impl<C: Context> ActiveValidator<C> {
         self.vidx == wunit.creator
     }
 
-    /// Returns whether a list of endorsements includes an endorsement created by a doppelganger.
-    /// An endorsement created by a doppelganger cannot be found in the local protocol state
-    /// (since we haven't created it ourselves).
-    pub(crate) fn includes_doppelgangers_endorsement(
-        &self,
-        endorsements: &Endorsements<C>,
-        state: &State<C>,
-    ) -> bool {
-        endorsements
-            .endorsers
-            .iter()
-            .any(|(vidx, _)| vidx == &self.vidx)
-            && !state.has_endorsement(endorsements.unit(), self.vidx)
+    /// Returns whether the incoming vertex was signed by our key even though we don't have it yet.
+    /// This can only happen if another node is running with the same signing key.
+    pub(crate) fn is_doppelganger_vertex(&self, vertex: &Vertex<C>, state: &State<C>) -> bool {
+        if !self.can_vote(state) {
+            // We don't have our own latest unit yet, so
```

Comment:
This comment ends abruptly, was it intentional?

Reply:
Sorry, my bad! I guess that was the point where I realized we have a problem with endorsements, and then I forgot to get back to it. :man_facepalming: 
@fizyk20: Feel free to just remove this comment for now.